libmachine/provision/systemd.go

```diff
@@ -22,7 +22,7 @@ func NewSystemdProvisioner(osReleaseID string, d drivers.Driver) SystemdProvisio
 		GenericProvisioner{
 			SSHCommander:      GenericSSHCommander{Driver: d},
 			DockerOptionsDir:  "/etc/docker",
-			DaemonOptionsFile: "/etc/systemd/system/docker.service",
+			DaemonOptionsFile: "/etc/systemd/system/docker.service.d/machine.conf",
```

Comment:
This has a side effect on the `redhat` provisioner as it currently verrides `GenerateDockerOptions` to override the system unit (rather than a drop-in) but references this var. I haven't tested yet https://github.com/bjaglin/machine/commit/ebe05412a09778f5f5e396c8c40fada29b556235, but simply removing `GenerateDockerOptions` looked promising.